client/app/templates/template-explorer.component.js

```diff
@@ -252,6 +271,53 @@ function ComponentController(ListView, TemplatesService, EventNotifications, $st
     if (angular.isUndefined(template.id)) {
       template = vm.selectedItemsList[0];
     }
-    $state.go('templates.editor', {templateId: template.id});
+    $state.go('templates.editor', { templateId: template.id });
+  }
+  function removeTemplate() {
+    let deleteTemplates;
+
+    if (vm.templatesToDelete.length > 0) {
+      deleteTemplates = vm.templatesToDelete;
+    } else if (vm.templatesToDelete) {
+      deleteTemplates = [vm.templatesToDelete];
+    }
+
+    if (deleteTemplates.length > 0) {
+      TemplatesService.deleteTemplates(deleteTemplates).then(removeSuccess,
+        removeFailure);
+    }
+    vm.confirmDelete = false;
+    vm.templatesToDelete = [];
+
+    function removeSuccess() {
```

Comment:
Confirmation of success or failure neeeeeeeeded :medal_military: 

(eventnotifications)